vms/platformvm/txs/fee/calculator_test.go

```diff
@@ -0,0 +1,816 @@
+
+				cfg := feeTestsDefaultCfg
+				upgrade := upgrade.Times{
+					DurangoTime:  durangoTime,
+					EUpgradeTime: eUpgradeTime,
+				}
+				return cfg, upgrade, chainTime
+			},
+			unsignedAndSignedTx: exportTx,
+			expectedError:       nil,
+			checksF: func(t *testing.T, fc *Calculator) {
+				require.Equal(t, fc.staticCfg.TxFee, fc.Fee)
+			},
+		},
+		{
+			description: "RewardValidatorTx pre EUpgrade",
+			cfgAndChainTimeF: func() (StaticConfig, upgrade.Times, time.Time) {
+				eUpgradeTime := time.Now().Truncate(time.Second)
+				chainTime := eUpgradeTime.Add(-1 * time.Second)
+
+				cfg := feeTestsDefaultCfg
+				upgrade := upgrade.Times{
+					DurangoTime:  durangoTime,
+					EUpgradeTime: eUpgradeTime,
+				}
+				return cfg, upgrade, chainTime
+			},
+			unsignedAndSignedTx: func(_ *testing.T) (txs.UnsignedTx, *txs.Tx) {
+				return &txs.RewardValidatorTx{
+					TxID: ids.GenerateTestID(),
+				}, nil
+			},
+			expectedError: nil,
+			checksF: func(t *testing.T, fc *Calculator) {
+				require.Equal(t, uint64(0), fc.Fee)
+			},
+		},
+		{
+			description: "AdvanceTimeTx pre EUpgrade",
+			cfgAndChainTimeF: func() (StaticConfig, upgrade.Times, time.Time) {
+				eUpgradeTime := time.Now().Truncate(time.Second)
+				chainTime := eUpgradeTime.Add(-1 * time.Second)
+
+				cfg := feeTestsDefaultCfg
+				upgrade := upgrade.Times{
+					DurangoTime:  durangoTime,
+					EUpgradeTime: eUpgradeTime,
+				}
+				return cfg, upgrade, chainTime
+			},
+			unsignedAndSignedTx: func(_ *testing.T) (txs.UnsignedTx, *txs.Tx) {
+				return &txs.AdvanceTimeTx{
+					Time: uint64(time.Now().Unix()),
+				}, nil
+			},
+			expectedError: nil,
+			checksF: func(t *testing.T, fc *Calculator) {
+				require.Equal(t, uint64(0), fc.Fee)
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.description, func(t *testing.T) {
+			cfg, upgrades, chainTime := tt.cfgAndChainTimeF()
+
+			uTx, _ := tt.unsignedAndSignedTx(t)
+			fc := NewStaticCalculator(cfg, upgrades, chainTime)
+
+			err := uTx.Visit(fc)
+			r.ErrorIs(err, tt.expectedError)
+			tt.checksF(t, fc)
+		})
+	}
+}
+
+func addValidatorTx(t *testing.T) (txs.UnsignedTx, *txs.Tx) {
+	r := require.New(t)
+
+	defaultCtx := snowtest.Context(t, snowtest.PChainID)
+
+	baseTx, stakes, _ := txsCreationHelpers(defaultCtx)
+	uTx := &txs.AddValidatorTx{
+		BaseTx: baseTx,
+		Validator: txs.Validator{
+			NodeID: defaultCtx.NodeID,
+			Start:  uint64(time.Now().Truncate(time.Second).Unix()),
+			End:    uint64(time.Now().Truncate(time.Second).Add(time.Hour).Unix()),
+			Wght:   feeTestDefaultStakeWeight,
+		},
+		StakeOuts: stakes,
+		RewardsOwner: &secp256k1fx.OutputOwners{
+			Locktime:  0,
+			Threshold: 1,
+			Addrs:     []ids.ShortID{ids.GenerateTestShortID()},
+		},
+		DelegationShares: reward.PercentDenominator,
+	}
+	sTx, err := txs.NewSigned(uTx, txs.Codec, feeTestSigners)
+	r.NoError(err)
+
+	return uTx, sTx
+}
+
+func addSubnetValidatorTx(t *testing.T) (txs.UnsignedTx, *txs.Tx) {
```

Comment:
I suspect we _might_ need this for dynamic fees (?) but since we don't need them in this PR and they introduce complexity I would move these into the dynamic PR diff. I want to avoid this if possible, which it looks like we are able to for now.